include/swift/AST/EvaluatorDependencies.h

```diff
@@ -99,6 +104,123 @@ inline DependencyScope getScopeForAccessLevel(AccessLevel l) {
 // edges in the incremental dependency graph invalidate entire files instead
 // of individual contexts.
 using DependencySource = llvm::PointerIntPair<SourceFile *, 1, DependencyScope>;
+
+/// A \c DependencyCollector is an aggregator of named references discovered in a
+/// particular \c DependencyScope during the course of request evaluation.
+struct DependencyCollector {
+private:
+  /// A stack of dependency sources in the order they were evaluated.
+  llvm::SmallVector<evaluator::DependencySource, 8> dependencySources;
+
+public:
+  DependencyCollector() = default;
+
+public:
+  /// Registers a named reference from the current dependency scope to a member
```

Comment:
@davidungar If you could look over these doc comments and make sure I've got this straight, that would be great.